tests/foreman/api/test_bookmarks.py

```diff
@@ -0,0 +1,290 @@
+from robottelo.constants import BOOKMARK_ENTITIES
+from robottelo.decorators import stubbed, tier1
+from robottelo.test import APITestCase
+
+
+class BookmarkTestCase(APITestCase):
+    """Test for common Bookmark operations via API"""
+
+    # CREATE TESTS
+    @stubbed()
+    @tier1
+    def test_positive_create_bookmark(self):
+        """Create a bookmark
+
+        @Feature: Scoped Search Bookmark Create
+
+        @Steps:
+
+        1. Create a bookmark with a random name and random valid controller
+        2. List the bookmarks
+
+        @Assert: No errors, Bookmark is listed, controller matches the
+        controller the bookmark was created for
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    @stubbed()
+    @tier1
+    def test_positive_create_bookmark_public(self):
+        """Create and check visibility of the (non)public bookmarks
+
+        @Feature: Scoped Search Bookmark Visibility
+
+        @Setup:
+
+        1. Create a non-admin user with 'viewer' role
+
+        @Steps:
+
+        1. Create a bookmark with a random name and public = true
+        2. List the bookmarks
+
+        @Assert: No errors, Bookmark is listed, controller matches the entity
+        the bookmark was created for and is displayed as public
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    @stubbed()
+    @tier1
+    def test_negative_create_bookmark_no_name(self):
+        """Create a bookmark with empty name
+
+        @Feature: Scoped Search Bookmark Create
+
+        @Steps:
+
+        1. Create a bookmark with a providing no name
+        2. List the bookmarks
+
+        @Assert: Error returned - name cannot be empty, Bookmark is not created
+        (not listed)
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    @stubbed()
+    @tier1
+    def test_negative_create_bookmark_no_querye(self):
+        """Create a bookmark with empty query
+
+        @Feature: Scoped Search Bookmark Create
+
+        @Steps:
+
+        1. Create a bookmark with a providing no query
+        2. List the bookmarks
+
+        @Assert: Error notification - search query cannot be empty, Bookmark is
+        not created (not listed)
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    @stubbed()
+    @tier1
+    def test_negative_create_bookmark_same_name(self):
+        """Create bookmarks with the same names
+
+        @Feature: Scoped Search Bookmark Create
+
+        @Setup:
+
+        1. Create a bookmark of a random name
+
+        @Steps:
+
+        1. Create a bookmark with providing the same name as the pre-created
```

Comment:
1. Create a new Bookmark using a random name
1. Create a second Bookmark, using the same name as the previous Bookmark. Assert that an error is raised
1. List Bookmarks. Assert that the Bookmark created is present and there's only one listed